src/web/tools/Vector.ts

```diff
@@ -0,0 +1,58 @@
+import { Directions } from '../../Directions';
+import PointerTracker from './PointerTracker';
+
+export default class Vector {
+  x: number = 0;
+  y: number = 0;
+  unitX: number = 0;
+  unitY: number = 0;
+
+  fromDirection(direction: Directions) {
+    [this.x, this.y] = [this.unitX, this.unitY] = DirectionTypeMappings.get(
+      direction
+    ) ?? [0, 0];
```

Comment:
```suggestion
[this.x, this.y] = [this.unitX, this.unitY] =
      DirectionTypeMappings.get(direction)!;
```

Assuming I haven't missed something, this value won't be undefined so we can use non-null assertion (correct me if I'm wrong).

Reply:
You are correct